processor/src/services/operation.service.ts

```diff
@@ -101,19 +102,20 @@ export class DefaultOperationService implements OperationService {
   }
 
   private async processPaymentModification(
+    payment: Payment,
     transactionType: string,
     pspReference: string,
     requestAmount: AmountSchemaDTO,
   ) {
     switch (transactionType) {
       case 'CancelAuthorization': {
-        return await this.operationProcessor.cancelPayment({ pspReference });
+        return await this.operationProcessor.cancelPayment({ pspReference, payment });
```

Comment:
same here and in the below methods, if we send the commercetools payment then we don't need to send the pspReference